packages/fether-react/src/stores/createAccountStore.js

```diff
@@ -28,37 +33,84 @@ export class CreateAccountStore {
     this.setName('');
   }
 
+  backupAccount = (address, password) => {
+    debug('Generating Backup JSON.');
+    return parityStore.api.parity
+      .exportAccount(address, password)
+      .then(res => {
+        const blob = new window.Blob([JSON.stringify(res)], {
+          type: 'application/json; charset=utf-8'
+        });
+
+        FileSaver.saveAs(blob, `${res.address}.json`);
+
+        return Promise.resolve('Successfully backed up account');
+      })
+      .catch(err => {
+        console.error(err);
+        return Promise.reject(err);
+      });
+  };
+
   generateNewAccount = () => {
     debug('Generating new account.');
     return this.setPhrase(null)
       .then(() => parityStore.api.parity.generateSecretPhrase())
       .then(this.setPhrase);
   };
 
-  saveAccountToParity = password => {
+  saveAccountToParity = async password => {
     debug('Saving account to Parity.');
-    return parityStore.api.parity
-      .newAccountFromPhrase(this.phrase, password)
-      .then(() =>
-        parityStore.api.parity.setAccountName(this.address, this.name)
-      )
-      .then(() =>
-        parityStore.api.parity.setAccountMeta(this.address, {
-          timestamp: Date.now()
-        })
-      );
+
+    try {
+      if (this.isJSON && this.address) {
+        // api.parity.newAccountFromWallet needs the address without prefix
```

Comment:
It will not be a string. When the file is dropped, callback receives stringified json data, immediately parses it, checks isFileValid, then the parsed json gets set as state and eventually to MobX Store. In `AccountImportOptions` lines 47-67.

Reply:
I meant: could you store the stringified data in the mobx store?

- callback gets stringified json
- you parse it, but only to check if its valid
- store the stringified version in mobx
- pass it down to newAccountFromWallet

Pro: the exact content from the json file is given to the node. Right now your parse it, and then stringify it again.

Reply:
Long story short, I need to parse it in order to append a `0x` prefix. 

Unfortunately, while the node expects the address without the prefix, and Parity keystore backups are without the prefix, everywhere else in the app (e.g. Blockie) expects the address with the prefix. 

So it ends up being either: save the parsed json in mobx while doing all the manipulations and finally only stringify in the api call, or save the stringified JSON and parse it each time it needs to be manipulated or need to grab a key (setJSON needs the parsed json). Neither are ideal but the first has fewer steps at least.

Reply:
My point is: the parsed JSON is used nowhere in the code.

The flow is: stringified JSON gets sent to node via newAccountFromWallet -> node parses it and creates an account -> `accountsInfo$` (from light.js, which does a pubsub of `parity_allAccountsInfo`) picks up that new account -> fether's code uses the accounts given by `accountsInfo$`.

If there's somewhere in the code which uses the parsed JSON, then imo it should be remove.d

Reply:
It does get used in the flow leading up to the flow you just outlined, in `createAccountStore => setJSON` for prefilling the account name and address ( not strictly necessary but is preferable UX-wise, particularly if the user imports an account through json and they're shown a blockie they don't recognize and the name field is blank, this is nerve wracking ). The parsed JSON gets used in extracting those values.

Reply:
Those two variables could be saved in `setName` and `setAddress`, right after parse.

Anyways, not a big deal, let's merge like this. My way was only an optimization improvement (memory + JSON.stringify operation).